tests/test_pages_layout.py

```diff
@@ -0,0 +1,133 @@
+import dash
+from dash import Dash, dcc, html
+import dash_labs as dl
+
+
+def get_app(path1="/", path2="/layout2"):
+    app = Dash(__name__, plugins=[dl.plugins.pages])
+
+    # test for storing arbitrary keyword arguments: An `id` prop is defined for every page
+    # test for defining multiple pages within a single file: layout is passed directly to `register_page`
+    # in the following two modules:
+    dash.register_page(
+        "multi_layout1",
+        layout=html.Div("text for multi_layout1", id="text_multi_layout1"),
+        path=path1,
+        title="Supplied Title",
+        description="This is the supplied description",
+        name="Supplied name",
+        image="birds.jpeg",
+        id="multi_layout1",
+    )
+    dash.register_page(
+        "multi_layout2",
+        layout=html.Div("text for multi_layout2", id="text_multi_layout2"),
+        path=path2,
+        id="multi_layout2",
+    )
+
+    app.layout = html.Div(
+        [
+            html.Div(
+                [
+                    html.Div(
+                        dcc.Link(
+                            f"{page['name']} - {page['path']}",
+                            id=page["id"],
+                            href=page["path"],
+                        )
+                    )
+                    for page in dash.page_registry.values()
+                ]
+            ),
+            dl.plugins.page_container,
+        ]
+    )
+    return app
+
+
+def test_pala001_layout(dash_duo):
+
+    dash_duo.start_server(get_app())
+
+    # test layout and title for each page in `page_registry`
+    for page in dash.page_registry.values():
+        dash_duo.find_element("#" + page["id"]).click()
+        dash_duo.wait_for_text_to_equal("#text_" + page["id"], "text for " + page["id"])
+        assert dash_duo.driver.title == page["title"], "check that page title updates"
+
+    # test redirects
+    dash_duo.wait_for_page(url="http://localhost:8050/v2")
+    dash_duo.wait_for_text_to_equal("#text_redirect", "text for redirect")
+    dash_duo.wait_for_page(url="http://localhost:8050/old-home-page")
+    dash_duo.wait_for_text_to_equal("#text_redirect", "text for redirect")
+
+    # test query strings
+    dash_duo.wait_for_page(url="http://localhost:8050/query-string?velocity=10")
+    assert (
+        dash_duo.find_element("#velocity").get_attribute("value") == "10"
+    ), "query string passed to layout"
+
+    # test path variables
+    dash_duo.wait_for_page(url="http://localhost:8050/a/none/b/none")
+    dash_duo.wait_for_text_to_equal("#path_vars", "variables from pathname:none none")
+
+    dash_duo.wait_for_page(url="http://localhost:8050/a/var1/b/var2")
+    dash_duo.wait_for_text_to_equal("#path_vars", "variables from pathname:var1 var2")
+
+    assert dash_duo.get_logs() == [], "browser console should contain no error"
+    # dash_duo.percy_snapshot("pala001_layout")
+
```

Comment:
could we add a test that includes both query strings and pathname variables?

also one with numbers, bools, and the literal string "none" just to lock down the behaviour.